davidpak/SmartNote
# Added export RPC, upload bugfixes, and test bugfixes

## What type of PR is this? (check all applicable)


## Description

- The `export` RPC has been partially implemented. Not all tests will pass (specifically `testExportRTF` and `testExportNotion`, these have not been implemented). This includes the `Exporter` interface and its related classes.
- `upload` had issues when detecting content type, which has been fixed.
- When the CI was added, some test classes had to be renamed as they were causing issues with the testing framework. This caused unresolved compiler errors.
- Added a dependency on a content type detector (Apache Tika).
- Refactored server entry point.

## QA Instructions, Screenshots, Recordings

See `INTERFACES.md` for information on the `export` RPC.

Internally, content is exported using an `Exporter`. An `Exporter` has a single `export` method whose job is to convert `String` input data, write it to a resource, and return the result info in a JSON object (compliant with the specification in `INTERFACES.md`). The exception `ExportException` has been added (along with subclasses) that indicates an issue with the underlying exporter (such as failing to connect to an API). An `Exporter` may be retrieved via the use of the `Exporters` class. See the inline documentation of the classes within `com.smartnote.server.export` for more information.

The following `Exporter` implementations have been added:

- `JSONExporter` - Exports to JSON, fully implemented.
- `RTFExporter` - Exports to RTF, not implemented.
- `NotionExporter` - Exports to Notion, not implemented.

The entry point of the server has been cleaned up. Different setup steps have been separated into different methods and now throw an `ExitEarlyEarlyException` if an early exit is desired. This exception stores a code that should be passed to `System.exit()`.

## Added/updated tests?
